index.js

```diff
@@ -325,6 +325,20 @@ function iterableEqual(leftHandOperand, rightHandOperand, options) {
       return false;
     }
   }
+
+  var leftHandProps = Object.keys(leftHandOperand);
+  var rightHandProps = Object.keys(rightHandOperand);
+  if (leftHandProps.length !== rightHandProps.length) {
+    return false;
+  }
+
+  for (var prop in rightHandProps) {
```

Comment:
Maybe we can do `for in` 


```js
return rightHandProps.every(function(prop) {
  return deepEqual(leftHandOperand[prop], rightHandOperand[prop], options)
})
```

to avoid `for in`